src/olympia/blocklist/utils.py

```diff
@@ -194,19 +201,27 @@ def save_versions_to_blocks(guids, submission):
         block.average_daily_users_snapshot = block.current_adu
         # And now update the BlockVersion instances - instances to add first
         block_versions_to_create = []
+        block_versions_to_update = []
+        is_soft = submission.block_type == BlockVersion.BLOCK_TYPE_CHOICES.SOFT_BLOCKED
         for version in block.addon_versions:
-            if version.id in submission.changed_version_ids and (
-                not change or not version.is_blocked
-            ):
-                block_version = BlockVersion(block=block, version=version)
-                block_versions_to_create.append(block_version)
+            if version.id in submission.changed_version_ids:
+                if version.is_blocked:
+                    block_version = version.blockversion
+                    block_versions_to_update.append(block_version)
+                else:
+                    block_version = BlockVersion(block=block, version=version)
+                    block_versions_to_create.append(block_version)
+                block_version.soft = is_soft
                 version.blockversion = block_version
```

Comment:
nit: you can move this into the `else` - in the `if ` we just set `block_version` to `version.blockversion`